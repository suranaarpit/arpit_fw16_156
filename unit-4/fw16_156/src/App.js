import logo from "./logo.svg";
import "./App.css";
import React from "react";
// import Day1 from './sprint-1/day-1/Day1';
// import Day1_2 from './sprint-1/day-1/Day1-2';
// import Day2 from "./sprint-1/day-2/Day2";
// import Day3 from "./sprint-1/day-3/Day3"
// import Sprint2Day2 from './sprint-2/day-2/Sprint2Day2';
// import Sprint2Day3 from "./sprint-2/day-3/Sprint2Day3";
// import Sprint3Day1 from "./sprint-3/day-1/Sprint3Day1";
// import Main from "./sprint-3/day-1/ReactCreditCard/Main";

function App() {
  return (
    <>
      {/* <Day1 ></Day1>}
      {<Day1_2 ></Day1_2> */}
      {/* <Day2></Day2> */}
      {/* <Day3></Day3> */}
      {/* <Sprint2Day2></Sprint2Day2> */}
      {/* <Sprint2Day3></Sprint2Day3> */}
      {/* <Sprint3Day1></Sprint3Day1> */}
      {/* <Main /> */}
    </>
  );
}

export default App;
