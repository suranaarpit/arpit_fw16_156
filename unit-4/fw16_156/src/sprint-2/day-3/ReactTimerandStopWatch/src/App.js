import React from "react";
import Main from "./Components/Imp";

function App() {
  return (
    <div>
      <Main />
    </div>
  );
}

export default App;
