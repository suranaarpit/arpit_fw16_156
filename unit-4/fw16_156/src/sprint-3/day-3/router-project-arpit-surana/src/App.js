import TechWorld from "./Components/TechWorld";


function App() {
  return (
    <div>
    <TechWorld/>
    </div>
  );
}

export default App;
