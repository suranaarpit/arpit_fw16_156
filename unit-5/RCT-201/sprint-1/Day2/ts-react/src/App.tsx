import Main from "./we/Components/Main";

function App() {
  return (
    <>
      <Main />
    </>
  );
}

export default App;
