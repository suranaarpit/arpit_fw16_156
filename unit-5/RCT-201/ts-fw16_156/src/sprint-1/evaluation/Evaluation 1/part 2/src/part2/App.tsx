import { useList } from "./hooks/useList";
import { useCounter } from "./hooks/useCounter";

function App() {
  // Entire thing bellow should work. you don't have to edit this file, only implement  hooks
  const [list, { push, pop, clear, reset, map }] = useList([1, 2, 3]);

  const { value, inc, dec, set } = useCounter(10);

  return (
    <div className="App">
      <button onClick={() => push(list.length + 1)}>Append</button>
      <button onClick={() => pop()}>pop</button>
      <button onClick={() => clear()}>clear</button>
      <button onClick={() => reset()}>reset</button>
      <button onClick={map}>Double every element</button>
      {list.map((l: any) => (
        <div key={l}>{l}</div>
      ))}

      <hr />
      <button onClick={() => inc(1)}>Add 1</button>
      <button onClick={() => inc(3)}>Add 3</button>
      <button onClick={() => dec(1)}>Sub 1</button>
      <button onClick={() => dec(4)}>Sub 4</button>
      <button onClick={set}>Set counter to 100</button>
      <h3>Value: {value}</h3>
    </div>
  );
}

export default App;
