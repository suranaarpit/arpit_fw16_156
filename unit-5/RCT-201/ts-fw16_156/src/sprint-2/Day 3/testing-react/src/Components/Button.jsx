import React from "react";
import "./btn.css";

const Button = ({ Varient, colorScheme, children, onClick }) => {
  return (
    <button data-testid="customBtn" onClick={onClick} className={`btn ${colorScheme || "red"} ${Varient || "ghost"}`}>
      {children}
    </button>
  );
};

export default Button;
