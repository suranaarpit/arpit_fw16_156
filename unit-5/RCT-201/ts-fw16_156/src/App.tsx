// import { Uth } from "./sprint-1/day-3/we/Components/uTimet";
import { GitHubUserSearch } from "./sprint-1/day-3/we/Components/gitUserSearch";
function App() {
  return (
    <>
      {/* <Uth /> */}
      <GitHubUserSearch/>
    </>
  );
}

export default App;
