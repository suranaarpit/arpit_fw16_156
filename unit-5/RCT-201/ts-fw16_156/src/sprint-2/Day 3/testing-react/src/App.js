import { useState } from "react";
import "./App.css";
import Button from "./Components/Button";

function App() {
  const [theme, setTheme] = useState("light");

  return (
    <div className="App">
      <h3>Theme is {theme}</h3>

      <Button
        onClick={() => {
          setTheme(theme === "light" ? "dark" : "light");
        }}
      >
       Change Theme
      </Button>
    </div>
  );
}

export default App;
