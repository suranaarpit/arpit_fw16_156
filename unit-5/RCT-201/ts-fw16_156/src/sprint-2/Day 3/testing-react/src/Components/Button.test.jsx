import { render, screen, fireEvent } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
import App from "../App";
import Button from "./Button";
import renderer from "react-test-renderer";

describe("Button Testing", function () {
  //   beforeAll(); => connect to a db
  //   beforeEach(); => //
  //   afterAll(); => disconnect from db
  //   afterEach(); => //

  it("Should have button in DOM", function () {
    render(<Button>Click Me</Button>); // Virtually
    // screen.debug();

    let button = screen.getByText("Click Me"); // document
    expect(button).toBeInTheDocument();
  });

  it("should render based on testID", function () {
    render(<Button>Click Me</Button>);
    let button = screen.getByTestId("customBtn");
    expect(button).toBeInTheDocument();
  });

  it("Should render empty Button", function () {
    render(<Button></Button>);

    let button = screen.getByTestId("customBtn");
    expect(button).toBeEmptyDOMElement();
  });

  //
  //   it.only("Should change Theme", function () {});
  it("Should change Theme", function () {
    render(<App></App>);

    let h3 = screen.getByText("Theme is light");
    // expect(h3).toBeInTheDocument();
    expect(h3).toHaveTextContent("light");

    let themeButton = screen.getByText("Change Theme");
    fireEvent.click(themeButton);
    // userEvent.click(themeButton);
    expect(h3).toHaveTextContent("dark");
  });
  it("should call given function", function () {
    const mockFn = jest.fn();
    render(<Button onClick={mockFn}>Click Me</Button>);
    // render(<Button onClick={mockFn}></Button>);

    // const btn = screen.getByTestId("customBtn");
    const btn = screen.getByText("Click Me");
    fireEvent.click(btn);
    fireEvent.click(btn);

    expect(mockFn).toBeCalledTimes(2);
  });

  // SnapShots => Screenshot
  it("Should match snapshot", function () {
    const tree = renderer
      .create(
        <Button colorScheme={"green"} Varient="ghost">
          Add Counter
        </Button>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Should show correct Green Classes", function () {
    render(<Button colorScheme={"green"}>Click Me</Button>);
    const btn = screen.getByTestId("customBtn");
    expect(btn).toHaveClass("green");
  });
});

// jest how to run single test suite "jest -t "Button Tests""
