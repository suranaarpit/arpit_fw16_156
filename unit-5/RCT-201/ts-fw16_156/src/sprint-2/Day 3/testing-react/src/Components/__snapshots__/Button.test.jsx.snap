// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Testing Should match snapshot 1`] = `
<button
  className="btn green ghost"
  data-testid="customBtn"
>
  Add Counter
</button>
`;
